### PR TITLE
[MNG-7848] Add s390x pipeline to Jenkins CI

### PR DESCRIPTION
**Pull request by kun-lu20**
JIRA issue: https://issues.apache.org/jira/browse/MNG-7848

Apache INFRA team has installed necessary JDK flavours on all s390x nodes (Please check issue https://issues.apache.org/jira/projects/INFRA/issues/INFRA-24781). We’d like to add the s390x build and test pipeline to Jenkins CI by adding `Jenkinsfile.s390x` to the code base.

Can anyone from Maven team help us review the PR and create the s390x pipeline on Jenkins? Thanks!

-------------------------------------------------------------------------------

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] SUMMARY`,
       where you replace `MNG-XXX` and `SUMMARY` with the appropriate JIRA issue.
 - [x] Also format the first line of the commit message like `[MNG-XXX] SUMMARY`.
       Best practice is to use the JIRA issue title in both the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by kun-lu20** [comment]
Hi @slawekjaranowski @cstamas @olamy ,

Could you please review this PR when you have a chance? Thank you very much!

**Comment by bmarwell** [comment]
@kun-lu20 why is this a new file? I think we could instead modify the existing pipeline, no?

**Comment by kun-lu20** [comment]
Thanks @cstamas !

**Comment by kun-lu20** [comment]
Hi @bmarwell ,

Thanks for your feedback! Yes, modifying the existing pipeline should be ok, but I think separating them also has some benefits:

1. We might not need to run the s390x pipeline as frequently as the amd64 pipeline, probably the nightly build is ok for s390x, like the s390x pipeline in other Apache projects, such as https://github.com/apache/zookeeper/blob/master/Jenkinsfile-s390x and https://github.com/apache/camel/blob/main/Jenkinsfile.s390x . 

2. For now we need to use a hard-coded mapped entry jdktool `ibm-semeru-8u362-b09` for JDK 8 on s390x, separating two pipelines will make the scripts clearer.

Please let me know what you think. Thanks again!

**Comment by slachiewicz** [comment]
Please run tests only on Maven 3.9 - to effectively use resources

**Comment by kun-lu20** [comment]
Hi @slachiewicz ,

Thanks for your feedback! I checked [amd64 Jenkinsfile](https://github.com/apache/maven/blob/master/Jenkinsfile) and it only uses Maven 3.8.x to build and test, the same as what we did in this PR for s390x Jenkinsfile:
```groovy
def buildMvn = '3.8.x'
...
def runITsMvn = '3.8.x'
```

Do you mean I should change the following line to maven-3.9.x only?
```groovy
if (env.BRANCH_NAME in ['master', 'maven-3.8.x', 'maven-3.9.x']){
```

Thank you very much!

**Comment by kun-lu20** [comment]
Hi @slachiewicz ,

I found that in the implementation of function `mvnFromVersion()` [here](https://github.com/apache/maven-jenkins-env/blob/61ee8f931e8ed3bf231970d4da21b0d62e58fcc2/vars/jenkinsEnv.groovy#L74-L75), tool `maven_3_latest` will be used for 3.8.x, so for now actually maven-3.9.3 is being used for build and test, both on amd64 and s390x Jenkinsfile.

Please let me know if I missed anything, thanks!

File: Jenkinsfile.s390x [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+properties([buildDiscarder(logRotator(artifactNumToKeepStr: '5', numToKeepStr: env.BRANCH_NAME=='master'?'5':'1'))])
+
+def buildOs = 'linux'
```
[the remark this answers is not in the document]
Comment (kun-lu20) [reply Jenkinsfile.s390x:22]: I think `linux` is ok in here, since we specified `s390x` in node declarations:

```groovy
node('s390x') {
```

**Comment by olamy** [comment]
who will maintain/monitor  the Jenkins job?
As we already have one and some GHA this is just adding one more job to watch and more noise (notifications)....
At least you could change the `  jenkinsNotify()` to send notifications to only you. As we already have enough.

**Comment by kun-lu20** [comment]
Hi @olamy ,

Our team (Linux on IBM Z Open Source Team) will maintain/monitor the Jenkins job.

Sure, I'll update the `Notifications` stage to send emails to us. I'll get back to you once the change is done.

Thanks!

**Comment by olamy** [comment]
@slachiewicz this shouldn’t have to be merged. As @kun-lu20 said in the comment, he will update the notifications to not send to us. The idea was to wait for this merge

**Comment by slachiewicz** [comment]
Do You know where we need to adjust it? I believe it's more at Jenkins pipeline level?

**Comment by kun-lu20** [comment]
Hi @olamy @slachiewicz ,

Thanks for merging this PR.

Yes, I think I know how to adjust it and I was working on that change (replacing `jenkinsNotify()` with another function which sends the notifications to our team).  Since this PR has been merged, I will raise a new PR for it ASAP.

**Comment by olamy** [comment]
> Hi @olamy @slachiewicz ,
> 
> Thanks for merging this PR.
> 
> Yes, I think I know how to adjust it and I was working on that change (replacing `jenkinsNotify()` with another function which sends the notifications to our team). Since this PR has been merged, I will raise a new PR for it ASAP.

correct you need to replace this with sending email to you

File: Jenkinsfile.s390x [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+node('s390x') {
+    dir('build') {
+        stage('Checkout') {
+            checkout scm
+        }
+
+        def WORK_DIR=pwd()
+        def MAVEN_GOAL='verify'
+
+        stage('Configure deploy') {
```
Comment (olamy) [inline Jenkinsfile.s390x:43]: this `deploy` goal should be remove.
thanks

**Pull request merged**